### PR TITLE
Bug 1861746: perform oauthaccesstokens only on upgraded clusters

### PR DESCRIPTION
**Pull request by soltysh**
/assign @deads2k @sanchezl

### CODE REVIEW

File: test/extended/cli/mustgather.go [hunk 1]
```diff
@@ -227,6 +227,12 @@ var _ = g.Describe("[sig-cli] oc adm must-gather", func() {
 						}
 						o.Expect(text).To(o.HavePrefix(`{"kind":"Event",`))
 						for _, token := range []string{"oauthaccesstokens", "oauthauthorizetokens", tokenName} {
```
Comment (sttts) [inline test/extended/cli/mustgather.go:229]: `// TODO(4.7): remove ethe whole oauthaccesstoken check`

Please also create a BZ.

File: test/extended/cli/mustgather.go [hunk 2]
```diff
@@ -227,6 +227,12 @@ var _ = g.Describe("[sig-cli] oc adm must-gather", func() {
...
+							// We can safely ignore 404 coming from apiserver reaching oauth.
+							// This is because oauth-apiserver is separate entity and this event is coming
+							// from openshift-apiserver trying to reach it.
+							if strings.Contains(text, `"username":"system:apiserver"`) && strings.Contains(text, `"code":404`) {
```
Comment (deads2k) [inline test/extended/cli/mustgather.go:233]: wait, when there is a 404, we get the token displayed?

Comment (soltysh) [reply test/extended/cli/mustgather.go]: Sample event where we fail:
```
{"kind":"Event","apiVersion":"audit.k8s.io/v1","level":"Metadata","auditID":"46af2b9c-afca-4c93-ab20-6b7c052db13c","stage":"ResponseComplete","requestURI":"/apis/oauth.openshift.io/v1/oauthaccesstokens/TEST","verb":"get","user":{"username":"system:apiserver","groups":["system:masters","system:authenticated"]},"sourceIPs":["::1","10.128.0.1"],"userAgent":"kube-apiserver/v1.19.0+e465e66 (linux/amd64) kubernetes/e465e66","objectRef":{"resource":"oauthaccesstokens","name":"TEST","apiGroup":"oauth.openshift.io","apiVersion":"v1"},"responseStatus":{"metadata":{},"status":"Failure","reason":"NotFound","code":404},"requestReceivedTimestamp":"2020-09-26T21:00:34.017891Z","stageTimestamp":"2020-09-26T21:00:34.020104Z","annotations":{"authorization.k8s.io/decision":"allow","authorization.k8s.io/reason":""}}
```
I spoke with @sttts and we're going to skip this test on fresh 4.6+ clusters, for older we'll keep it.

File: test/extended/cli/mustgather.go [hunk 3]
```diff
@@ -226,8 +235,14 @@ var _ = g.Describe("[sig-cli] oc adm must-gather", func() {
 							continue // ignore truncated data
 						}
 						o.Expect(text).To(o.HavePrefix(`{"kind":"Event",`))
-						for _, token := range []string{"oauthaccesstokens", "oauthauthorizetokens", tokenName} {
-							o.Expect(text).NotTo(o.ContainSubstring(token))
+
+						// - if we're on 4.7+, don't check at all
```
Comment (sttts) [inline test/extended/cli/mustgather.go:239]: `// TODO(4.7): remove this check`.

**Pull request merged**